probe/integration-tests/src/test/java/org/jboss/weld/probe/integration/tests/ProbeBeansTest.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Test
+    public void testBeanDetail() throws IOException {
+        JsonObject modelBeanDetail = getBeanDetail(BEANS_PATH_ALL, ModelBean.class, url);
+        assertEquals(ModelBean.class.getName(), modelBeanDetail.getString(BEAN_CLASS));
+        assertEquals(BeanType.MANAGED.name(), modelBeanDetail.getString(KIND));
+        assertEquals("@" + RequestScoped.class.getSimpleName(), modelBeanDetail.getString(SCOPE));
+
+        // check @Model bean
+        JsonArray qualifiers = modelBeanDetail.getJsonArray(QUALIFIERS);
+        JsonArray types = modelBeanDetail.getJsonArray(TYPES);
+        JsonArray dependencies = modelBeanDetail.getJsonArray(DEPENDENCIES);
+        assertTrue(checkStringInArrayRecursively(Default.class.getSimpleName(), null, qualifiers, false));
+        assertTrue(checkStringInArrayRecursively(ModelBean.class.getName(), null, types, false));
+        assertTrue(checkStringInArrayRecursively(SessionScopedBean.class.getName(), BEAN_CLASS, dependencies, false));
+
+        // check sessionscoped bean
+        JsonObject sessionScopedBeanDetail = getBeanDetail(BEANS_PATH_ALL, SessionScopedBean.class, url);
+        assertEquals(SessionScopedBean.class.getName(), sessionScopedBeanDetail.getString(BEAN_CLASS));
+        assertEquals(BeanType.MANAGED.name(), sessionScopedBeanDetail.getString(KIND));
+        assertEquals("@" + SessionScoped.class.getSimpleName(), sessionScopedBeanDetail.getString(SCOPE));
+
+        types = sessionScopedBeanDetail.getJsonArray(TYPES);
+        JsonArray dependents = sessionScopedBeanDetail.getJsonArray(DEPENDENTS);
+        assertTrue(checkStringInArrayRecursively(SessionScopedBean.class.getName(), null, types, false));
+        assertTrue(checkStringInArrayRecursively(Serializable.class.getName(), null, types, false));
+        assertTrue(checkStringInArrayRecursively(ModelBean.class.getName(), BEAN_CLASS, dependents, false));
+
+        // check interceptor detail
+        JsonObject testInterceptorDetail = getBeanDetail(BEANS_PATH_ALL, TestInterceptor.class, url);
+        assertEquals(TestInterceptor.class.getName(), testInterceptorDetail.getString(BEAN_CLASS));
+        assertEquals(BeanType.INTERCEPTOR.name(), testInterceptorDetail.getString(KIND));
+        assertEquals("@" + Dependent.class.getSimpleName(), testInterceptorDetail.getString(SCOPE));
+
+    }
+
+    @Test
+    public void testProducers() throws IOException {
+        JsonObject allBeans = getPageAsJSONObject(BEANS_PATH_ALL, url);
+        JsonArray beansData = allBeans.getJsonArray(DATA);
+
+        List<JsonObject> producers = getAllJsonObjectsByClass(TestProducer.class, beansData);
+
+        Optional<JsonObject> fieldProducer = producers.stream().filter((JsonObject o) -> o.getString(KIND).equals(BeanType.PRODUCER_FIELD.name()))
+                .findAny();
+        assertNotNull("Cannot find producer field from " + TestProducer.class.getName(), fieldProducer.get());
```

Comment:
This will never fail with the "Cannot find producer ..." message. It will either pass or throw NoSuchElementException if the optional is empty.